frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1335,6 +1392,23 @@ pub mod pallet {
 	#[pallet::getter(fn minimum_untrusted_score)]
 	pub type MinimumUntrustedScore<T: Config> = StorageValue<_, ElectionScore>;
 
+	/// Rolling average of the [`MinimumUntrustedScore`] up to the previous
+	/// `MinimumUntrustedScoreUpdateInterval` elections.
+	///
+	/// The first index refers to how many successful elections have passed since the current
+	/// rolling average calculation started. The second item keeps track of the interval that the
+	/// current average has been calculated with, so that we can detect if the interval value has
+	/// changed and reset the average.
+	#[pallet::storage]
+	pub type MinimumUntrustedScoreAvg<T: Config> = StorageValue<_, (u32, u32, ElectionScore)>;
```

Comment:
That is basically what the first and last elements of the tuple do. The other element keeps track of the `MinimumUntrustedScoreUpdateInterval` that the current avg has been calculated with, so that we can reset the average when the interval changes mid-rolling average calculation.